src/java/org/apache/cassandra/locator/AbstractEndpointSnitch.java

```diff
@@ -55,6 +56,11 @@ public int compare(InetAddress a1, InetAddress a2)
         });
     }
 
+    public Snapshot getSnapshot(InetAddress endpoint)
```

Comment:
What are your thoughts on making this optional as well? I usually don't like controlling program flow via exceptions, but also see that this makes sense in isolation